lib/Serialization/ModuleFile.cpp

```diff
@@ -117,6 +117,11 @@ ModuleFile::ModuleFile(std::shared_ptr<const ModuleFileSharedCore> core)
   allocateBuffer(Identifiers, core->Identifiers);
 }
 
+bool ModuleFile::isAllowModuleWithCompilerErrorsEnabled() const {
+  return allowedCompilerErrors() &&
+         (!FileContext || getContext().LangOpts.AllowModuleWithCompilerErrors);
```

Comment:
Could there be a scenario where the reader doesn't accept errors and the module being read did accept errors?

Reply:
There shouldn't be, though that makes me think of the other case - ie. a module that has errors but wasn't compiled with allowing errors. That *should* never happen, but it seems reasonable that we'd just want to continue in that case.